internal/mode/static/handler_test.go

```diff
@@ -214,6 +221,45 @@ var _ = Describe("eventHandler", func() {
 		})
 	})
 
+	It("should set the health checker status properly when there are changes", func() {
+		e := &events.UpsertEvent{Resource: &v1beta1.HTTPRoute{}}
+		batch := []interface{}{e}
+
+		fakeProcessor.ProcessReturns(true, &graph.Graph{})
+
+		Expect(handler.cfg.healthChecker.readyCheck(nil)).ToNot(Succeed())
+		handler.HandleEventBatch(context.Background(), batch)
+		Expect(handler.cfg.healthChecker.readyCheck(nil)).To(Succeed())
+	})
+
+	It("should set the health checker status properly when there are no changes or errors", func() {
+		e := &events.UpsertEvent{Resource: &v1beta1.HTTPRoute{}}
+		batch := []interface{}{e}
+
+		Expect(handler.cfg.healthChecker.readyCheck(nil)).ToNot(Succeed())
+		handler.HandleEventBatch(context.Background(), batch)
+		Expect(handler.cfg.healthChecker.readyCheck(nil)).To(Succeed())
+	})
+
+	It("should set the health checker status properly when there is an error", func() {
+		e := &events.UpsertEvent{Resource: &v1beta1.HTTPRoute{}}
+		batch := []interface{}{e}
+
+		fakeProcessor.ProcessReturns(true, &graph.Graph{})
+		fakeNginxRuntimeMgr.ReloadReturns(errors.New("reload error"))
+
+		handler.HandleEventBatch(context.Background(), batch)
+
+		Expect(handler.cfg.healthChecker.readyCheck(nil)).ToNot(Succeed())
+
+		// now send an update with no changes; should still return an error
+		fakeProcessor.ProcessReturns(false, &graph.Graph{})
+
+		handler.HandleEventBatch(context.Background(), batch)
+
+		Expect(handler.cfg.healthChecker.readyCheck(nil)).ToNot(Succeed())
```

Comment:
shall we further test here how readyCheck eventually Succeeds?

Reply:
That's tested in the two tests above, it would be redundant to do it again here.

Reply:
No that's true, I can add that.